go/mcap/indexed_message_iterator.go

```diff
@@ -254,16 +393,23 @@ func readRecord(r io.Reader) (OpCode, []byte, error) {
 	return opcode, record, nil
 }
 
-func (it *indexedMessageIterator) Next(_ []byte) (*Schema, *Channel, *Message, error) {
-	if !it.hasReadSummarySection {
-		err := it.parseSummarySection()
-		if err != nil {
+// Next2 yields the next message from the iterator, re-using the allocations from `msg` if
+// provided.
```

Comment:
re-use only occurs if the message data buffer has sufficient room. Maybe we should note that for completeness here.